ee/watchdog/controller_windows.go

```diff
@@ -180,224 +191,426 @@ func (wc *WatchdogController) ServiceEnabledChanged(enabled bool) {
+
+	repetition := repetitionObj.ToIDispatch()
+	defer repetition.Release()
+
+	// set the repetition interval. PT30M=30 minutes
+	if _, err = oleutil.PutProperty(repetition, "Interval", "PT30M"); err != nil {
+		return fmt.Errorf("setting time trigger interval: %w", err)
+	}
+
+	// begin creation of the task action
+	actionsProp, err := oleutil.GetProperty(taskDefinition, "Actions")
+	if err != nil {
+		return fmt.Errorf("getting actions property: %w", err)
+	}
+
+	actions := actionsProp.ToIDispatch()
+	defer actions.Release()
+
+	// see action types here https://learn.microsoft.com/en-us/windows/win32/api/taskschd/ne-taskschd-task_action_type
+	// 0=TASK_ACTION_EXEC
+	execActionTemplate, err := oleutil.CallMethod(actions, "Create", uint(0))
+	if err != nil {
+		return fmt.Errorf("creating event action: %w", err)
 	}
 
-	wc.slogger.Log(ctx, slog.LevelInfo, "completed watchdog service installation")
+	execAction := execActionTemplate.ToIDispatch()
+	defer execAction.Release()
+
+	installedExePath, err := getExecutablePath(identifier)
+	if err != nil {
+		return fmt.Errorf("determining watchdog executable path: %w", err)
+	}
+
+	if _, err = oleutil.PutProperty(execAction, "Path", `"`+installedExePath+`"`); err != nil {
+		return fmt.Errorf("setting action path: %w", err)
+	}
+
+	taskArgs := fmt.Sprintf(`watchdog -config "%s"`, configFilePath)
+	if _, err = oleutil.PutProperty(execAction, "Arguments", taskArgs); err != nil {
+		return fmt.Errorf("setting action arguments: %w", err)
+	}
+
+	// now register the task!
+	_, err = oleutil.CallMethod(rootFolder, "RegisterTaskDefinition",
+		taskName,       // Task name
+		taskDefinition, // Task definition
+		uint(6),        // Flags: 6=TASK_CREATE_OR_UPDATE see https://learn.microsoft.com/en-us/windows/win32/api/taskschd/ne-taskschd-task_creation
+		"SYSTEM",       // User: run as system
+		nil,            // password (nil for the current user, we expect this installed from SYSTEM)
+		uint(5),        // 5=TASK_LOGON_SERVICE_ACCOUNT see https://learn.microsoft.com/en-us/windows/win32/api/taskschd/ne-taskschd-task_logon_type
+		nil,            // SDDL (security descriptor definition language string, nil for our purposes here)
+	)
+
+	if err != nil {
+		return fmt.Errorf("registering task definition: %w", err)
+	}
 
 	return nil
 }
 
-// RemoveService utilizes the passed serviceManager to remove any existing watchdog service if it exists
-func RemoveService(serviceManager *mgr.Mgr) error {
-	existingService, err := serviceManager.OpenService(launcherWatchdogServiceName)
+// RemoveWatchdogTask will determine the task name based on the given identifier, and remove
+// the task from the scheduler service. This is exported for use by our remote uninstallation paths
+func RemoveWatchdogTask(identifier string) error {
+	if strings.TrimSpace(identifier) == "" {
+		identifier = launcher.DefaultLauncherIdentifier
+	}
+
+	taskName := launcher.TaskName(identifier, watchdogTaskType)
+	// init COM
+	ole.CoInitialize(0)
```

Comment:
Can you add a comment here and the other couple places that you call `CoInitialize` in this file that we discard the error returned by `CoInitialize` because it (harmlessly) returns `S_FALSE` if we call it more than once?

Reply:
will do!